openmdao/core/group.py

```diff
@@ -941,7 +951,9 @@ def solve_linear(self, dumat, drmat, vois, mode=None, solver=None):
         if len(rhs_buf) == 0:
             return
 
+        solver.rel_inputs = rel_inputs
         sol_buf = solver.solve(rhs_buf, self, mode=mode)
+        solver.rel_inputs = None
```

Comment:
i would argue that it seems like doing it this way obfuscates a necessary part of that function call, making the code less robust..